WrightTools/data/_data.py

```diff
@@ -1834,65 +1834,97 @@ def split(self, axis, positions, units='same',
         # process ---------------------------------------------------------------------------------
-        outs = []
+        outs = wt_collection.Collection(name='split', parent=parent)
         start = 0
-        stop = -1
+        stop = 0
         for i in range(len(indicies) + 1):
             # get start and stop
-            start = stop + 1  # previous value
+            start = stop  # previous value
             if i == len(indicies):
-                stop = len(axis[:])
+                stop = len(axis) 
             else:
-                stop = indicies[i]
+                stop = indicies[i] + 1
             # new data object prepare
-            new_data = self.copy()
-            # axis of interest will be FIRST
-            transpose_order = range(len(new_data.axes))
-            # replace axis_index with zero
-            transpose_order = [0 if i == axis_index else i for i in transpose_order]
-            transpose_order[0] = axis_index
-            new_data.transpose(transpose_order, verbose=False)
-            # axis
-            new_data.axes[0][:] = new_data.axes[0][start:stop + 1]
-            # channels
-            for channel in new_data.channels:
-                channel[:] = channel[start:stop + 1]
-            # transpose out
-            new_data.transpose(transpose_order, verbose=False)
-            outs.append(new_data)
+            new_name = "split%03d"%i
+            if stop - start < 1:
+                outs.create_data("")
+            elif stop - start == 1:
+                attrs = dict(self.attrs)
+                attrs.pop('name', None)
+                attrs.pop('axis_names', None)
+                attrs.pop('channel_names', None)
+                attrs.pop('constant_names', None)
+
+                new_data = outs.create_data(new_name, **attrs)
+                #for c in self.constants:
+                #   new_data.create_constant(c.natural_name, c.value, c.units)
+                #new_data.create_constant(axis.natural_name, axis[start], axis.units)
+
+                for ax in self.axes:
+                    if ax != axis:
+                        attrs = dict(ax.attrs)
+                        attrs.pop('name', None)
+                        attrs.pop('units', None)
+                        new_data.create_axis(ax.natural_name, ax[:], ax.units,  **attrs)
+
+                slc = [slice(None)] * len(self.shape)
+                slc[axis_index] = start
+                for ch in self.channels:
+                    attrs = dict(ch.attrs)
+                    attrs.pop('name', None)
+                    attrs.pop('units', None)
+                    new_data.create_channel(ch.natural_name, ch[:][slc], ch.units, **attrs)
+            else:
+                attrs = dict(self.attrs)
+                attrs.pop('name', None)
+                attrs.pop('axis_names', None)
+                attrs.pop('channel_names', None)
+                attrs.pop('constant_names', None)
```

Comment:
I agree, we should pop `axis_names`, `channel_names` and `constant_names` within `Data.__init__` (and similar)